Client/src/Screens/Mainpage.jsx

```diff
@@ -0,0 +1,145 @@
+              </button>
+
+              {/* Sidebar Links */}
+              <div className="flex md:flex-col items-center">
+                <Link
+                  to="/saved"
+                  className="mb-5 flex flex-col items-center"
+                  onClick={() => {
+                    setIsSpecialView(true);
+                    setIsMenuOpen(false);
+                  }}
+                >
+                  <img src={iconCheck} alt="Saved" className="md:h-[4.8rem] md:w-[3.9rem] custom-xss:h-[3rem] custom-xss:w-[2.8rem] custom-xss:mt-0.1" />
+                </Link>
+                <Link
+                  to="/templates"
+                  className="mb-4 flex flex-col items-center"
+                  onClick={() => {
+                    setIsSpecialView(true);
+                    setIsMenuOpen(false);
+                  }}
+                >
+                  <img src={iconEnvelope} alt="Templates" className="md:h-[5rem] md:w-[5.5rem] custom-xss:h-[3.3rem] custom-xss:w-[4rem] custom-xss:mb-[.3rem] custom-xss:ml-3 md:mr-3 md:ml-[.8rem]" />
+                </Link>
+                <Link
+                  to="/download"
+                  className="mb-6 flex flex-col items-center"
+                  onClick={() => {
+                    setIsSpecialView(true);
+                    setIsMenuOpen(false);
+                  }}
+                >
+                  <img src={iconDownload} alt="Download" className="md:h-[5rem] md:w-[5.5rem] ml-1 custom-xss:h-[3rem] custom-xss:w-[4.3rem] custom-xss:mt-[.4rem] custom-xss:ml-3 md:mr-1" />
+                </Link>
+                <Link
+                  to="/settings"
+                  className="flex flex-col items-center"
+                  onClick={() => {
+                    setIsSpecialView(true);
+                    setIsMenuOpen(false);
+                  }}
+                >
+                  <img src={iconCog} alt="Settings" className="md:h-[5rem] md:w-[6.1rem] sm:w-[4rem] custom-xss:h-[3rem] custom-xss:w-[4.3rem] custom-xss:mb-[0.9rem]" />
+                </Link>
+              </div>
+            </div>
+
+            <div className="w-full md:w-[47.5%] bg-gray-500 flex-1"></div>
+          </>
+        ) : (
+          <div className="flex-1">
+            <Routes>
+              <Route path="/saved" element={<Saved />} />
+              <Route path="/templates" element={<Templates />} />
+              <Route path="/download" element={<Download />} />
+              <Route path="/settings" element={<Settings />} />
+            </Routes>
+          </div>
+        )}
+      </div>
+
+      {/* Footer */}
+      <footer className="bg-blue-200 text-xl flex justify-center py-5 font-bold">
+        <img src={iconFooter} alt="Footer" className="h-6 w-[10rem] custom-xss:h-[0.7rem] custom-xss:w-[6rem] md:w-[13rem] md:h-[1.3rem]" />
+      </footer>
+    </div>
+  );
+}
+
```

Comment:
All these other pages should be separate jsx files... don't have multiple components defined in the same page